[control-plane-manager][docs] Update documentation about managing K8s versions (#3719)
deckhouse/deckhouse: modules/040-control-plane-manager/docs/README.md

## Version control

> **Patch versions** of control plane components (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.

Author: ```suggestion
**Patch versions** of control plane components (i.e. within the minor version, for example, from `1.23.3` to `1.23.5`) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.
```

## Version control

**Patch versions** of control plane components (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.

Upgrading **minor versions** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be managed using the [`kubernetesVersion`](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. It specifies the automatic update mode (if set to `Automatic`) or the desired minor version of the control plane. The default control plane version (to use with `kubernetesVersion: Automatic`) as well as a list of supported Kubernetes versions can be found in [the documentation](../../supported_versions.html#kubernetes).

The control plane upgrade is performed in a safe way for both single-master and multi-master clusters. The API server may be temporarily unavailable during the upgrade. At the same time, it does not affect the operation of applications in the cluster and can be performed without scheduling a maintenance window.

> If the version specified for upgrading (the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current version of control plane in the cluster, then a smart strategy for changing component versions is launched:

Reviewer: ```suggestion
If the target version (set in the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current control plane version in the cluster, a smart strategy for changing component versions is applied:
```

- General remarks
  - Updating in different NodeGroups is performed in parallel. Within each NodeGroup, nodes are updated sequentially, one at a time.
- When upgrading:
  - Upgrades are carried out sequentially, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
  - At each step, the control plane version is upgraded first, followed by kubelet upgrades on the cluster nodes. 

>   - On top of that, the maximum version increment is one at a time; that is, the new minor version can differ from the lowest kubelet version on the nodes by 1 at max.

Author: ```suggestion
```

## Version control

> The update of the **patch version** of the control plane components (i.e. within the minor version, for example, from 1.24.3 to 1.24.8) occurs automatically together with the 
> update of the Deckhouse version. You can't manage patch version updates.

Author: ```suggestion
**Patch versions** of control plane components (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.
```

> Updating the **minor version** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be controlled using the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. In the `kubernetesVersion` parameter, you can specify the automatic update mode (value `Automatic`) or the desired minor version of the control plane. The version of control plane that is used by default (with `kubernetesVersion: Automatic`), as well as a list of supported versions of Kubernetes can be found in the [documentation](../../supported_versions.html#kubernetes).

Author: ```suggestion
Upgrading the **minor version** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be controlled using the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. In the `kubernetesVersion` parameter, you can specify the automatic update mode (value `Automatic`) or the desired minor version of the control plane. The version of control plane that is used by default (with `kubernetesVersion: Automatic`), as well as a list of supported versions of Kubernetes can be found in the [documentation](../../supported_versions.html#kubernetes).
```

The control plane updgrade is performed in a safe way for both single-master and multi-master clusters. During the upgrade, there may be short-term unavailability of the API server. The update does not affect the operation of applications in the cluster and can be performed without allocating a maintenance window.

If the version specified for upgrading (the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current version of control plane in the cluster, then a smart strategy for changing component versions is launched:
- General remarks
  - Updating in different NodeGroups is performed in parallel. Within each NodeGroup, nodes are updated sequentially, one at a time.
- When upgrading:

>   - The upgrading is performed in sequential steps, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.

Reviewer 2: `The upgrade` or `upgrading` ?

Author: Replace 'updating' with 'upgrading'.

I will as @kirkonru to help with the other suggestion

>   - At each step, the control plane version is updated first, followed by the kubelet update on the cluster nodes. 
>   - The update at each step occurs a maximum of one version higher than the minimum version of kubelet on the nodes.

Reviewer: ```suggestion
  - At each step, the control plane version is upgraded first, followed by kubelet upgrades on the cluster nodes. 
  - On top of that, the maximum version increment is one at a time; that is, the new minor version can differ from the lowest kubelet version on the nodes by 1 at max.
```

Reviewer: It seems to me that line 70 is not much different from line 68.

- When downgrading:

>   - A successful downgrade is guaranteed only one version down from the maximum minor version of the control plane ever used in the cluster.
>   - The kubelet is downgraded on the cluster nodes first, followed by the control plane components downgrade.

Reviewer: ```suggestion
  - Successful downgrading is only guaranteed for a single version down from the maximum minor version of the control plane ever used in the cluster.
  - kubelets on the cluster nodes are downgraded first, followed by the control plane components.
```
